packages/edit-site/src/components/page-pages/index.js

```diff
@@ -48,17 +49,50 @@ export default function PagePages() {
 		hiddenFields: [ 'date', 'featured-image' ],
 		layout: {},
 	} );
-	// Request post statuses to get the proper labels.
+
 	const { records: statuses } = useEntityRecords( 'root', 'status' );
-	const postStatuses = useMemo(
-		() =>
-			statuses === null
-				? EMPTY_OBJECT
-				: Object.fromEntries(
-						statuses.map( ( { slug, name } ) => [ slug, name ] )
-				  ),
-		[ statuses ]
-	);
+	const { postStatuses, defaultStatuses } = useMemo( () => {
+		return {
+			postStatuses:
+				statuses === null
+					? EMPTY_OBJECT
+					: Object.fromEntries(
+							statuses.map( ( { slug, name } ) => [ slug, name ] )
+					  ),
+			defaultStatuses:
+				statuses === null
+					? DEFAULT_STATUSES
+					: statuses
+							.filter( ( { slug } ) => slug !== 'trash' )
+							.map( ( { slug } ) => slug )
+							.join(),
+		};
+	}, [ statuses ] );
+
+	useEffect( () => {
```

Comment:
I feel like this is a bit overkill, at least for now. Do you feel strongly about adding it now?

Reply:
We do need to update the view upon receiving the statuses, otherwise it'd have stale data. For example: what if a site has a different set of statuses from the ones we hard-coded at `DEFAULT_STATUSES`?

The only thing we could remove is the `DEFAULT_STATUSES` to `defaultStatuses` check. I think it is still worth maintaining: it's just a single line of code, and it prevents an unnecessary double request.

Reply:
>The only thing we could remove is the DEFAULT_STATUSES to defaultStatuses

That was what I was referring to, sorry for not clarifying. I mean it's useful to avoid the request, but the block with the comments it's big 😅 . Let's simplify this part then just a bit.

How about assign `DEFAULT_STATUSES` with the sorted values(so no need to split->sort->join), and also sort `defaultStatuses` in the `useMemo`. So the comparison here is just for two strings.

Reply:
Done this at https://github.com/WordPress/gutenberg/pull/55476/commits/78c0609f32d409c3d6fcf53d59d3100207067696

Note that I had to add another comment:

```js
// DEFAULT_STATUSES is intentionally sorted. Items do not have spaces in between them.
// The reason for that is to match defaultStatuses because we compare both strings below (see useEffect).
const DEFAULT_STATUSES = 'draft,future,pending,private,publish'; // All statuses but 'trash'.
```

With the previous approach, I wanted to cover against future humans (including myself) modifying ever slightly that constant, with the consequence of breaking the comparison. Perhaps I should be more optimistic about humans :)